runtime/src/bridge.rs

```diff
@@ -58,9 +62,21 @@ decl_module! {
         fn deposit_event() = default;
 
         /// Transfers some amount of the native token to some recipient on a (whitelisted) destination chain.
+        /// Adds additional fee to compensate the current cost of target chains
         #[weight = SimpleDispatchInfo::FixedNormal(1_000_000)]
         pub fn transfer_native(origin, amount: BalanceOf<T>, recipient: Vec<u8>, dest_id: chainbridge::ChainId) -> DispatchResult {
             let source = ensure_signed(origin)?;
+			let token_fee = <T as pallet_balances::Trait>::Balance::from(TOKEN_FEE);
+			let total_amount = U256::from(amount.saturated_into()) + U256::from(token_fee.saturated_into());
+
+			// Ensure account has enough balance for both fee and transfer
+			// Check to avoid balance errors down the line that leave balance storage in an inconsistent state
+			let current_balance = T::Currency::free_balance(&source);
```

Comment:
Main reason of this check is to avoid burning tokens from the account and not being able to succeed bridging the token. Only applicable to errors related to balance.

Reply:
Agree that it is not strictly needed since we don't modify state before doing the burn_fee and the transfer.

Reply:
The state is changed and the operations within are not reverted. So for example, if the user has enough balance for the fee but not for the transfer, indeed it will error out in the second one, but the first fee will be lost already.

Reply:
Yes, but I think we should follow Substrate's principle to not optimize for UX in the runtime. Sending a tx costs tx fees, even if it aborts/reverts. Loosing those fees would not be much different from that.

The UI can check for the balance before sending the tx.

Reply:
Can we move all the checks before the state is changed?

Reply:
If the fee was relatively small I would agree with it, but we are talking at 20 RAD. I can move all the checks before the state change, I thought I did, but I missed the whitelist one, thanks for pointing it out.